### ISSUE TITLE
Can't use certain items with "if player's tool is" or "if player's offhand tool is"

### ISSUE THREAD

**Issue by Zaphara94** [SkriptLang/Skript#5022]
### Skript/Server Version

```markdown
[09:45:31 INFO]: Skript version 2.6.3
[09:45:31 INFO]: Customize Minecraft's mechanics with simple scripts written in plain English sentences.
[09:45:31 INFO]: Website: https://skriptlang.github.io/Skript
[09:45:31 INFO]: Authors: Njol, Mirreski, SkriptLang Team and Contributors
[09:46:37 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[09:46:37 INFO]: [Skript] Skript's documentation can be found here: https://skriptlang.github.io/Skript
[09:46:37 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[09:46:37 INFO]: [Skript] Server Version: git-Paper-118 (MC: 1.19.2)
[09:46:37 INFO]: [Skript] Skript Version: 2.6.3
[09:46:37 INFO]: [Skript] Installed Skript Addons: 
[09:46:37 INFO]: [Skript]  - skript-placeholders v1.5.2 (https://github.com/APickledWalrus/skript-placeholders)
[09:46:37 INFO]: [Skript]  - skRayFall v1.9.28 (https://sk.rayfall.net/)
[09:46:37 INFO]: [Skript]  - SkBee v1.17.3 (https://github.com/ShaneBeee/SkBee)
[09:46:37 INFO]: [Skript]  - WolvSK v1.7.2
[09:46:37 INFO]: [Skript] Installed dependencies: 
[09:46:37 INFO]: [Skript]  - Vault v1.7.3-b131
[09:46:37 INFO]: [Skript]  - WorldGuard v7.0.8-beta-01+cbb2ba7
```

### Bug Description

```v
on place of beacon:
	if off hand item is lingering potion or sponge or dragon head or totem:
		set {_beacon.player} to nbt of off hand item
		if off hand item is sponge:
			set {_nbt.beacon.player} to "{Levels:1,Primary:4}"
		else if off hand item is totem of undying:
			set {_nbt.beacon.player} to "{Levels:1,Primary:31}"
		else if off hand item is dragon head:
			set {_nbt.beacon.player} to "{Levels:1,Primary:32}"

on right click:
	if player's target is horse:
		if player's tool is end rod:
			set {_horse.%player%.nbt} to nbt of player's target
```

These two sections of code, which used to work before, now fail and disable the scripts that they were a part of with the error "Line 188: (SilkSpawners.sk)
Can't compare a slot with particle
Line: else if off hand item is totem of undying:"

So far, these are the only two items that have this issue, and we have disabled a few scripts, and sections of other scripts, to initialize all our scripts.

### Expected Behavior

The scripts were supposed to initialize without error, and allow us to check what items a player is holding for certain things. 

### Steps to Reproduce

This happens no matter which alias I use for the items in question, they keep saying that the item is a particle.

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by TheLimeGlass** [comment]
Skript doesn't have NBT.

**Comment by Zaphara94** [comment]
If i replace either “end rod” or “totem of undying” with “torch” the problem goes away, and works as intended. And another script i have that denies lava-flow (it literally is just “on flow: if event-block is lava: cancel event”) errors with the same error: “can’t compare a block with a particle”. Again if i change this from lava to water, it works fine. And these just broke when we updated to the lates 1.19 version.

**Comment by AyhamAl-Ali** [comment]
Probably duplicate of #4773

**Comment by AyhamAl-Ali** [comment]
Can't replicate, please test the same code without any addons (comment out the NBT stuff) and let us know if that issue is still

**Comment by Fusezion** [comment]
Decided to test it for them on my server I'm getting the same error as they are
when using `is end rod` you'll get the error but changing to `is ender rod` it'll work as intended

---------
![image](https://user-images.githubusercontent.com/81952476/185817598-62ae4539-1a8b-4e16-9d47-d42967bd15ae.png)
![image](https://user-images.githubusercontent.com/81952476/185817613-884dbe51-199e-4f28-bb65-756effaae26f.png)

**Comment by AyhamAl-Ali** [comment]
> Decided to test it for them on my server I'm getting the same error as they are when using `is end rod` you'll get the error but changing to `is ender rod` it'll work as intended

Your case is duplicate of #4773.

I will be closing this issue due to lack of response and most likely an addon issue however if it turns out to be Skript's fault a team member can reopen this.
